### PR TITLE
Fixes system culture dependent Regex test

### PR DESCRIPTION
**Pull request by AlexejLiebenthal**
Fixes test bug: #33789

### CODE REVIEW

File: src/System.Text.RegularExpressions/tests/Regex.Cache.Tests.cs [hunk 1]
```diff
@@ -98,7 +98,7 @@ public void Ctor_Cache_Uses_culture_and_options()
                 Assert.True(Regex.IsMatch("1", "1", RegexOptions.IgnoreCase));
                 Assert.True(Regex.IsMatch("1", "1", RegexOptions.Multiline));
                 Assert.True(GetCachedItemsNum() == 2);
-                CultureInfo.CurrentCulture = CultureInfo.GetCultureInfo("de-DE");
+                CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
```
[the remark this answers is not in the document]
Comment (AlexejLiebenthal) [reply src/System.Text.RegularExpressions/tests/Regex.Cache.Tests.cs]: I think it shouldn't lose test coverage as this test shouldn't test ``CultureInfo``. Or in other words the getter and setter of ``CultureInfo.CurrentCulture`` should have been tested in their own test cases.

I also already thought about your suggestion implementing it, with a ternary operator, get the current culture and then set a different one. But from your issue #25082, my assumption was, that all build machines are running on "en-US" or "es-ES" culture and anyone elses local computer setup has a culture different from ``InvariantCulture``. And so I implemented it this way to not introduce more logic in this already quiet complex test.

But as I see now in the CI, the OSX build is probably running on an ``InvariantCulture``, so the implementation must be quiet more complex.

What do you think about to change the line like this with an explaining comment:
```cs
// Force to set a different culture than the current culture!
CultureInfo.CurrentCulture = CultureInfo.CurrentCulture.Equals(CultureInfo.GetCultureInfo("de-DE")) ?
                                CultureInfo.InvariantCulture) : 
                                CultureInfo.GetCultureInfo("de-DE");
```

[the remark this answers is not in the document]
Comment (danmosemsft) [reply src/System.Text.RegularExpressions/tests/Regex.Cache.Tests.cs]: Looks good to me

**Comment by AlexejLiebenthal** [comment]
@dotnet-bot test UWP CoreCLR x64 Debug Build

**Comment by danmosemsft** [comment]
Thanks @AlexejLiebenthal ! Do you want to take another? 
https://github.com/dotnet/corefx/issues?q=is%3Aopen+is%3Aissue+label%3Aup-for-grabs

**Pull request merged**